extensions/s3/src/main/java/io/deephaven/extensions/s3/S3Instructions.java

```diff
@@ -99,6 +110,28 @@ public Credentials credentials() {
         return Credentials.defaultCredentials();
     }
 
+    /**
+     * The size of each part (in MiB) to upload when writing to S3, defaults to {@value #DEFAULT_PART_SIZE_MB} MiB. The
+     * minimum allowed part size is {@value #MIN_PART_SIZE_MB} MiB. Setting a higher value may increase throughput, but
+     * may also increase memory usage. Note that the maximum number of parts allowed for a single file is 10,000.
+     * Therefore, for {@value #DEFAULT_PART_SIZE_MB} MiB part size, the maximum size of a single file that can be
+     * written is {@value #DEFAULT_PART_SIZE_MB} * 10,000 MiB.
+     */
+    @Default
+    public int partSizeMib() {
```

Comment:
I'm not the biggest fan of assuming units for the user; IMO, this should just be `long partSize()`, _unless_ the underlying library itself is configured for int `MiB`.